### PR TITLE
Add wesnoth.sync module for the synchronization-related commands

### PR DESCRIPTION
**Pull request by CelticMinstrel**

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
I'll do the requested name change tonight.

File: src/scripting/game_lua_kernel.cpp [hunk 1]
```diff
@@ -4266,6 +4264,21 @@ game_lua_kernel::game_lua_kernel(game_state & gs, play_controller & pc, reports
...
 	lua_setfield(L, -2, "audio");
 	lua_pop(L, 1);
+	
+	// Create the sync module
+	cmd_log_ << "Adding sync module...\n";
+	static luaL_Reg const sync_callbacks[] {
+		{ "invoke_command",    &intf_invoke_synced_command },
+		{ "run_unsynced",      &intf_do_unsynced },
+		{ "evaluate",          &intf_synchronize_choice },
+		{ "multi_evaluate",    &intf_synchronize_choices },
```
Comment (CelticMinstrel) [inline src/scripting/game_lua_kernel.cpp:4274]: @Pentarctagon Should it be `evaluate` and `evaluate_multiple` or should the first one be `evaluate_single`?

Comment (Pentarctagon) [reply src/scripting/game_lua_kernel.cpp:4274]: `evaluate_multiple` and `evaluate_single` sounds good.

File: src/scripting/game_lua_kernel.cpp [hunk 2]
```diff
@@ -2704,7 +2704,9 @@ namespace
 			lua_pushnumber(L, side);
 			if (luaW_pcall(L, 1, 1, false)) {
 				if(!luaW_toconfig(L, -1, cfg)) {
-					lua_kernel_base::get_lua_kernel<game_lua_kernel>(L).log_error("function returned to wesnoth.synchronize_choice a table which was partially invalid");
+					static const char* msg = "function returned to wesnoth.sync.[multi_]evaluate a table which was partially invalid";
```
Comment (stevecotton) [inline src/scripting/game_lua_kernel.cpp:2707]: Name has changed.

**Pull request merged**